static/js/stream_list.js

```diff
@@ -157,6 +159,29 @@ export function build_stream_list(force_rerender) {
     $parent.append(elems);
 }
 
+export function build_add_stream_link() {
+    // Only append when parent does not have children
+
+    const more_streams = stream_data.get_non_default_stream_names();
+    const num_more_streams = more_streams.length;
+    const $parent = $("#add-stream-link");
+
+    $parent.empty();
+
+    // child textNode to append to anchor element
+    let text_content = `Borrow ${num_more_streams} more streams`;
+
+    const icon_elem = '<i class="fa fa-plus-circle" aria-hidden="true"></i>';
+    let anchor_elem = `<a href="#streams/all">${icon_elem}${text_content}</a>`;
+
+    if (num_more_streams < 1) {
+        text_content = "Create a stream";
+        anchor_elem = `<a id="add-stream-link-create">${icon_elem}${text_content}</a>`;
+    }
+
+    $parent.append(anchor_elem);
```

Comment:
Let's do this login in a Handlebars template; we avoid doing HTML construction like this in JavaScript.

See also our guide on internationalization: https://zulip.readthedocs.io/en/latest/translating/internationalization.html -- all strings must be properly tagged for translation.